v-next/hardhat/src/internal/builtin-plugins/solidity-test/task-action.ts

```diff
@@ -110,13 +120,33 @@ const runSolidityTests: NewTaskActionFunction<TestActionArguments> = async (
   let includesErrors = false;
 
   const solidityTestConfig = hre.config.solidityTest;
+  let observabilityConfig: ObservabilityConfig | undefined;
+  if (hre.globalOptions.coverage) {
+    assertHardhatInvariant(
+      hre instanceof HardhatRuntimeEnvironmentImplementation,
+      "Expected HRE to be an instance of HardhatRuntimeEnvironmentImplementation",
+    );
+
+    observabilityConfig = {
+      codeCoverage: {
+        onCollectedCoverageCallback: async (coverageData: Uint8Array[]) => {
```

Comment:
Can you talk me through this callback? What does it add to the coverage output?

Reply:
Sure! This is a callback that EDR calls when a specific tag (which identifies a code section that can be covered) is hit. You'll notice that we get an array of tags as an argument, that's to make it more efficient.

When we receive the tags, we add them to the coverage manager where we just store them at that stage. The tag processing part is initiated after all the tests are done.